src/fp/list.nim

```diff
@@ -114,6 +114,24 @@ proc foldRightF*[T, U](xs: List[T], z: () -> U, f: (T, () -> U) -> U): U =
   if xs.isEmpty: z()
   else: f(xs.head, () => xs.tail.foldRightF(z, f))
 
+proc unfoldLeft*[T](f: T -> Option[(T, T)], x:T): List[T] {. inline .}=
```

Comment:
This seems to be an instance of another Nim issue, which would hit its birthday in a week, A possible workaround could be something like this:
```nim
proc unfoldLeft*[T, U](f: U -> Option[(item: T, seed: U)], x:U): List[T] {. inline .}=
```
This should work because of `Nim` tuple coercion.